facilitator/src/idl.rs

```diff
@@ -112,65 +162,38 @@ impl BatchSignature {
         let mut batch_header_signature = None;
         let mut key_identifier = None;
 
-        for tuple in record {
-            match (tuple.0.as_str(), tuple.1) {
+        for (field_name, field_value) in fields {
+            match (field_name.as_str(), field_value) {
                 ("batch_header_signature", Value::Bytes(v)) => batch_header_signature = Some(v),
                 ("key_identifier", Value::String(v)) => key_identifier = Some(v),
-                (f, _) => {
```

Comment:
Hmm... I made this change purposefully. Otherwise, we can't introduce a new field without it being a breaking change, i.e. we'd have to update every facilitator that might read this message type simultaneously. (or introduce the field as a two-release process: in the first release, allow reading the new field; in the second release, start writing the new field. that's a lot of pain just for a new field!)

I should've called this out beforehand as a purposeful behavior change (apologies)--if my argument is not convincing, I'll revert this part of the change.

Reply:
(not that this should be taken as iron-clad evidence, but "allowing but ignoring unknown fields" is how e.g. protobuf is implemented, for pretty much the reason I described above)

Reply:
No, this is fine. TBH I don't expect these Avro messages to ever change anyway.